e2e-tests/desktop/tests/targets.spec.js

```diff
@@ -166,13 +166,18 @@ test.describe('Targets tests', () => {
       await authedPage
         .getByRole('link', { name: targetWithTwoHosts.name })
         .click();
-      await authedPage.getByRole('button', { name: 'Connect' }).click();
 
-      await expect(
-        authedPage.getByRole('heading', { name: 'Choose a Host' }),
-      ).toBeVisible();
-
-      await authedPage.getByRole('button', { name: host }).click();
+      // eslint-disable-next-line playwright/no-conditional-in-test
+      if (host === 'Quick Connect') {
+        await authedPage.getByRole('button', { name: host }).click();
+      } else {
+        await authedPage
+          .getByRole('table')
+          .getByRole('cell', { name: host })
+          .locator('..')
```

Comment:
We should avoid using `.locator('..')` when possible, there's usually other ways of getting the right locator as using xpath is a last resort.

You might have been able to do something like this (I didn't confirm this):

```
authedPage
  .getByRole('row')
  .filter({ hasText: host})
  .getByRole('button', { name: 'Connect' })
  .click();
```

Reply:
Nice